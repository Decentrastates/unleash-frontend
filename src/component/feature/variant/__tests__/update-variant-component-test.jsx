import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core';

import UpdateVariant from './../update-variant-component';
import renderer from 'react-test-renderer';
import { weightTypes } from '../enums';
import theme from '../../../../themes/main-theme';
import { Provider } from 'react-redux';
import { createStore } from 'redux';

jest.mock(
    '../AddVariant/OverrideConfig/OverrideConfig.jsx',
    () => 'OverrideConfig'
);

const mockStore = {
    uiConfig: {
        toJS: () => ({
            flags: {
                P: true,
            },
        }),
    },
};

const mockReducer = state => state;

test('renders correctly with without variants', () => {
    const tree = renderer.create(
        <Provider store={createStore(mockReducer, mockStore)}>
            <ThemeProvider theme={theme}>
                <MemoryRouter>
                    <UpdateVariant
                        key={0}
                        variants={[]}
                        addVariant={jest.fn()}
                        removeVariant={jest.fn()}
                        updateVariant={jest.fn()}
                        stickinessOptions={['default']}
                        updateStickiness={jest.fn()}
                        editable
                    />
                </MemoryRouter>
            </ThemeProvider>
        </Provider>
    );

    expect(tree).toMatchSnapshot();
});

test('renders correctly without variants and no permissions', () => {
    const tree = renderer.create(
        <Provider store={createStore(mockReducer, mockStore)}>
            <ThemeProvider theme={theme}>
                <MemoryRouter>
                    <UpdateVariant
                        key={0}
                        variants={[]}
                        addVariant={jest.fn()}
                        removeVariant={jest.fn()}
                        updateVariant={jest.fn()}
                        stickinessOptions={['default']}
                        updateStickiness={jest.fn()}
                        editable
                    />
                </MemoryRouter>
            </ThemeProvider>
        </Provider>
    );

    expect(tree).toMatchSnapshot();
});

test('renders correctly with with variants', () => {
    const featureToggle = {
        name: 'toggle.variants',
        description: 'description',
        enabled: false,
        strategies: [
            {
                name: 'gradualRolloutRandom',
                parameters: {
                    percentage: 50,
                },
            },
        ],
        variants: [
            {
                name: 'blue',
                weight: 34,
                overrides: [
                    {
                        field: 'userId',
                        values: ['1337', '123'],
                    },
                ],
            },
            {
                name: 'yellow',
                weight: 33,
            },
            {
                name: 'orange',
                weight: 33,
                weightType: weightTypes.FIX,
                payload: {
                    type: 'string',
                    value: '{"color": "blue", "animated": false}',
                },
            },
        ],
        createdAt: '2018-02-04T20:27:52.127Z',
    };
    const tree = renderer.create(
        <Provider store={createStore(mockReducer, mockStore)}>
            <ThemeProvider theme={theme}>
                <MemoryRouter>
                    <UpdateVariant
                        key={0}
                        variants={featureToggle.variants}
                        addVariant={jest.fn()}
                        removeVariant={jest.fn()}
                        updateVariant={jest.fn()}
                        stickinessOptions={['default']}
                        updateStickiness={jest.fn()}
                        editable
                    />
                </MemoryRouter>
            </ThemeProvider>
        </Provider>
    );

    expect(tree).toMatchSnapshot();
});
