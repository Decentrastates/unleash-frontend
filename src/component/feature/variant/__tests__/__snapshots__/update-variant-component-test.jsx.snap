// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with with variants 1`] = `
<section
  style={
    Object {
      "maxWidth": "700px",
      "padding": "16px",
    }
  }
>
  <p>
    Variants allows you to return a variant object if the feature toggle is considered enabled for the current request. When using variants you should use the
     
    <code
      style={
        Object {
          "color": "navy",
        }
      }
    >
      getVariant()
    </code>
     method in the Client SDK.
  </p>
  <table
    className="mdl-data-table mdl-shadow--2dp variantTable"
  >
    <thead>
      <tr>
        <th>
          Variant name
        </th>
        <th
          className="labels"
        />
        <th>
          Weight
        </th>
        <th
          className="actions"
        />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          onClick={[Function]}
        >
          blue
        </td>
        <td
          className="labels"
        >
           
          <react-mdl-Chip
            style={
              Object {
                "backgroundColor": "rgba(173, 216, 230, 0.2)",
              }
            }
          >
            Overrides
          </react-mdl-Chip>
        </td>
        <td>
          3.4
           %
        </td>
        <td
          className="actions"
        >
          <react-mdl-IconButton
            name="edit"
            onClick={[Function]}
          />
          <react-mdl-IconButton
            name="delete"
            onClick={[Function]}
          />
        </td>
      </tr>
      <tr>
        <td
          onClick={[Function]}
        >
          yellow
        </td>
        <td
          className="labels"
        >
           
        </td>
        <td>
          3.3
           %
        </td>
        <td
          className="actions"
        >
          <react-mdl-IconButton
            name="edit"
            onClick={[Function]}
          />
          <react-mdl-IconButton
            name="delete"
            onClick={[Function]}
          />
        </td>
      </tr>
      <tr>
        <td
          onClick={[Function]}
        >
          orange
        </td>
        <td
          className="labels"
        >
          <react-mdl-Chip>
            Payload
          </react-mdl-Chip>
           
        </td>
        <td>
          3.3
           %
        </td>
        <td
          className="actions"
        >
          <react-mdl-IconButton
            name="edit"
            onClick={[Function]}
          />
          <react-mdl-IconButton
            name="delete"
            onClick={[Function]}
          />
        </td>
      </tr>
    </tbody>
  </table>
  <br />
  <p>
    <a
      href="#add-variant"
      onClick={[Function]}
      title="Add variant"
    >
      Add variant
    </a>
  </p>
  <form
    onSubmit={[Function]}
  >
    <react-mdl-Dialog
      className="modal"
      open={false}
    >
      <react-mdl-DialogTitle
        style={
          Object {
            "padding": "10px",
          }
        }
      />
      <react-mdl-DialogContent
        style={
          Object {
            "minHeight": "350px",
            "padding": "10px",
          }
        }
      >
        <p
          style={
            Object {
              "color": "red",
            }
          }
        />
        <react-mdl-Textfield
          floatingLabel={true}
          label="Variant name"
          name="name"
          onChange={[Function]}
          placeholder=""
          style={
            Object {
              "width": "100%",
            }
          }
          type="name"
        />
        <br />
        <br />
        <react-mdl-Tooltip
          className="tooltip"
          label={
            <span>
              Passed to the variant object. 
              <br />
              Can be anything (json, value, csv)
            </span>
          }
        >
          <p
            style={
              Object {
                "marginBottom": "0",
              }
            }
          >
            <strong>
              Payload 
            </strong>
            <react-mdl-Icon
              name="info"
            />
          </p>
        </react-mdl-Tooltip>
        <react-mdl-Grid
          noSpacing={true}
        >
          <react-mdl-Cell
            col={3}
          >
            <div
              className="mdl-textfield mdl-js-textfield mdl-textfield--floating-label is-dirty is-upgraded"
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <select
                className="mdl-textfield__input"
                name="type"
                onChange={[Function]}
                style={
                  Object {
                    "width": "auto",
                  }
                }
                value="string"
              >
                <option
                  value="string"
                >
                  string
                </option>
                <option
                  value="json"
                >
                  json
                </option>
                <option
                  value="csv"
                >
                  csv
                </option>
              </select>
              <label
                className="mdl-textfield__label"
                htmlFor="textfield-conextName"
              >
                Type
              </label>
            </div>
          </react-mdl-Cell>
          <react-mdl-Cell
            col={9}
          >
            <react-mdl-Textfield
              floatingLabel={true}
              label="Value"
              name="value"
              onChange={[Function]}
              rows={1}
              style={
                Object {
                  "width": "100%",
                }
              }
              value=""
            />
          </react-mdl-Cell>
        </react-mdl-Grid>
        <a
          href="#add-override"
          onClick={[Function]}
        >
          <small>
            Add override
          </small>
        </a>
      </react-mdl-DialogContent>
      <react-mdl-DialogActions>
        <react-mdl-Button
          colored={true}
          raised={true}
          type="submit"
        >
          Save
        </react-mdl-Button>
        <react-mdl-Button
          onClick={[Function]}
          type="button"
        >
          Cancel
        </react-mdl-Button>
      </react-mdl-DialogActions>
    </react-mdl-Dialog>
  </form>
</section>
`;

exports[`renders correctly with without variants 1`] = `
<section
  style={
    Object {
      "maxWidth": "700px",
      "padding": "16px",
    }
  }
>
  <p>
    Variants allows you to return a variant object if the feature toggle is considered enabled for the current request. When using variants you should use the
     
    <code
      style={
        Object {
          "color": "navy",
        }
      }
    >
      getVariant()
    </code>
     method in the Client SDK.
  </p>
  <table
    className="mdl-data-table mdl-shadow--2dp variantTable"
  >
    <thead>
      <tr>
        <th>
          Variant name
        </th>
        <th
          className="labels"
        />
        <th>
          Weight
        </th>
        <th
          className="actions"
        />
      </tr>
    </thead>
    <tbody />
  </table>
  <br />
  <p>
    <a
      href="#add-variant"
      onClick={[Function]}
      title="Add variant"
    >
      Add variant
    </a>
  </p>
  <form
    onSubmit={[Function]}
  >
    <react-mdl-Dialog
      className="modal"
      open={false}
    >
      <react-mdl-DialogTitle
        style={
          Object {
            "padding": "10px",
          }
        }
      />
      <react-mdl-DialogContent
        style={
          Object {
            "minHeight": "350px",
            "padding": "10px",
          }
        }
      >
        <p
          style={
            Object {
              "color": "red",
            }
          }
        />
        <react-mdl-Textfield
          floatingLabel={true}
          label="Variant name"
          name="name"
          onChange={[Function]}
          placeholder=""
          style={
            Object {
              "width": "100%",
            }
          }
          type="name"
        />
        <br />
        <br />
        <react-mdl-Tooltip
          className="tooltip"
          label={
            <span>
              Passed to the variant object. 
              <br />
              Can be anything (json, value, csv)
            </span>
          }
        >
          <p
            style={
              Object {
                "marginBottom": "0",
              }
            }
          >
            <strong>
              Payload 
            </strong>
            <react-mdl-Icon
              name="info"
            />
          </p>
        </react-mdl-Tooltip>
        <react-mdl-Grid
          noSpacing={true}
        >
          <react-mdl-Cell
            col={3}
          >
            <div
              className="mdl-textfield mdl-js-textfield mdl-textfield--floating-label is-dirty is-upgraded"
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <select
                className="mdl-textfield__input"
                name="type"
                onChange={[Function]}
                style={
                  Object {
                    "width": "auto",
                  }
                }
                value="string"
              >
                <option
                  value="string"
                >
                  string
                </option>
                <option
                  value="json"
                >
                  json
                </option>
                <option
                  value="csv"
                >
                  csv
                </option>
              </select>
              <label
                className="mdl-textfield__label"
                htmlFor="textfield-conextName"
              >
                Type
              </label>
            </div>
          </react-mdl-Cell>
          <react-mdl-Cell
            col={9}
          >
            <react-mdl-Textfield
              floatingLabel={true}
              label="Value"
              name="value"
              onChange={[Function]}
              rows={1}
              style={
                Object {
                  "width": "100%",
                }
              }
              value=""
            />
          </react-mdl-Cell>
        </react-mdl-Grid>
        <a
          href="#add-override"
          onClick={[Function]}
        >
          <small>
            Add override
          </small>
        </a>
      </react-mdl-DialogContent>
      <react-mdl-DialogActions>
        <react-mdl-Button
          colored={true}
          raised={true}
          type="submit"
        >
          Save
        </react-mdl-Button>
        <react-mdl-Button
          onClick={[Function]}
          type="button"
        >
          Cancel
        </react-mdl-Button>
      </react-mdl-DialogActions>
    </react-mdl-Dialog>
  </form>
</section>
`;

exports[`renders correctly with without variants and no permissions 1`] = `
<section
  style={
    Object {
      "maxWidth": "700px",
      "padding": "16px",
    }
  }
>
  <p>
    Variants allows you to return a variant object if the feature toggle is considered enabled for the current request. When using variants you should use the
     
    <code
      style={
        Object {
          "color": "navy",
        }
      }
    >
      getVariant()
    </code>
     method in the Client SDK.
  </p>
  <table
    className="mdl-data-table mdl-shadow--2dp variantTable"
  >
    <thead>
      <tr>
        <th>
          Variant name
        </th>
        <th
          className="labels"
        />
        <th>
          Weight
        </th>
        <th
          className="actions"
        />
      </tr>
    </thead>
    <tbody />
  </table>
  <br />
  <form
    onSubmit={[Function]}
  >
    <react-mdl-Dialog
      className="modal"
      open={false}
    >
      <react-mdl-DialogTitle
        style={
          Object {
            "padding": "10px",
          }
        }
      />
      <react-mdl-DialogContent
        style={
          Object {
            "minHeight": "350px",
            "padding": "10px",
          }
        }
      >
        <p
          style={
            Object {
              "color": "red",
            }
          }
        />
        <react-mdl-Textfield
          floatingLabel={true}
          label="Variant name"
          name="name"
          onChange={[Function]}
          placeholder=""
          style={
            Object {
              "width": "100%",
            }
          }
          type="name"
        />
        <br />
        <br />
        <react-mdl-Tooltip
          className="tooltip"
          label={
            <span>
              Passed to the variant object. 
              <br />
              Can be anything (json, value, csv)
            </span>
          }
        >
          <p
            style={
              Object {
                "marginBottom": "0",
              }
            }
          >
            <strong>
              Payload 
            </strong>
            <react-mdl-Icon
              name="info"
            />
          </p>
        </react-mdl-Tooltip>
        <react-mdl-Grid
          noSpacing={true}
        >
          <react-mdl-Cell
            col={3}
          >
            <div
              className="mdl-textfield mdl-js-textfield mdl-textfield--floating-label is-dirty is-upgraded"
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <select
                className="mdl-textfield__input"
                name="type"
                onChange={[Function]}
                style={
                  Object {
                    "width": "auto",
                  }
                }
                value="string"
              >
                <option
                  value="string"
                >
                  string
                </option>
                <option
                  value="json"
                >
                  json
                </option>
                <option
                  value="csv"
                >
                  csv
                </option>
              </select>
              <label
                className="mdl-textfield__label"
                htmlFor="textfield-conextName"
              >
                Type
              </label>
            </div>
          </react-mdl-Cell>
          <react-mdl-Cell
            col={9}
          >
            <react-mdl-Textfield
              floatingLabel={true}
              label="Value"
              name="value"
              onChange={[Function]}
              rows={1}
              style={
                Object {
                  "width": "100%",
                }
              }
              value=""
            />
          </react-mdl-Cell>
        </react-mdl-Grid>
        <a
          href="#add-override"
          onClick={[Function]}
        >
          <small>
            Add override
          </small>
        </a>
      </react-mdl-DialogContent>
      <react-mdl-DialogActions>
        <react-mdl-Button
          colored={true}
          raised={true}
          type="submit"
        >
          Save
        </react-mdl-Button>
        <react-mdl-Button
          onClick={[Function]}
          type="button"
        >
          Cancel
        </react-mdl-Button>
      </react-mdl-DialogActions>
    </react-mdl-Dialog>
  </form>
</section>
`;
